packages/overlay/src/Overlay.ts

```diff
@@ -0,0 +1,156 @@
+    TriggerInteractionsV1,
+} from './overlay-types.js';
+import { Placement } from '@floating-ui/dom/src/types.js';
+import { VirtualTrigger } from './VirtualTrigger.js';
+import { reparentChildren } from '@spectrum-web-components/shared';
+
+const supportsPopover = 'showPopover' in document.createElement('div');
+
+let OverlayFeatures = OverlayDialog(OverlayBase);
+if (supportsPopover) {
+    OverlayFeatures = OverlayPopover(OverlayFeatures);
+} else {
+    OverlayFeatures = OverlayNoPopover(OverlayFeatures);
+}
+
+type OverlayOptions = {
+    delayed?: boolean;
+    notImmediatelyClosable?: boolean;
+    offset?: number | [number, number]; // supporting multi-axis
+    placement?: Placement;
+    receivesFocus?: 'auto' | 'true' | 'false';
+    trigger?: HTMLElement | VirtualTrigger;
+    type?: 'modal' | 'page' | 'hint' | 'auto' | 'manual';
+};
+
+/**
+ * @element sp-overlay
+ *
+ * @fires sp-opened - announces that an overlay has completed any entry animations
+ * @fires sp-closed - announce that an overlay has compelted any exit animations
+ */
+export class Overlay extends OverlayFeatures implements OverlayBase {
+    public static update(): void {
+        const overlayUpdateEvent = new CustomEvent('sp-update-overlays', {
+            bubbles: true,
+            composed: true,
+            cancelable: true,
+        });
+        document.dispatchEvent(overlayUpdateEvent);
+    }
+
+    public static async open(
+        target: HTMLElement,
+        interaction: TriggerInteractionsV1,
+        content: HTMLElement,
+        options: OverlayOptionsV1
+    ): Promise<() => void>;
+    public static async open(
+        content: HTMLElement,
+        options?: OverlayOptions
+    ): Promise<Overlay>;
+    public static async open(
+        targetOrContent: HTMLElement,
+        interactionOrOptions:
+            | TriggerInteractionsV1
+            | OverlayOptions
+            | undefined,
+        content?: HTMLElement,
+        options?: OverlayOptionsV1
+    ): Promise<Overlay | (() => void)> {
+        const v2 = arguments.length === 2;
+        const overlayContent = content || targetOrContent;
+        const overlay = new Overlay();
+        overlay.dispose = () => {
+            overlay.addEventListener('sp-closed', () => {
+                if (!restored) {
+                    restoreContent();
+                    restored = true;
+                }
+                requestAnimationFrame(() => {
+                    overlay.remove();
+                });
+            });
+            overlay.open = false;
+            overlay.dispose = noop;
+        };
+        let restored = false;
+        const restoreContent = reparentChildren([overlayContent], overlay, {
+            position: 'beforeend',
+            prepareCallback: (el) => {
+                const slot = el.slot;
+                el.removeAttribute('slot');
+                return () => {
+                    el.slot = slot;
+                };
+            },
```

Comment:
My naive assumption was that only the v2 branch would reparent nodes - this (or the top of this file) might be a good place for a 1-paragraph overview of the mechanisms of how Overlay works under the hood.

Is the purpose of this to make sure that if overlaid content is projected into a light DOM slot, that it's not projected when it's in the overlay container (so it's visible outside of its "normal" slot environment), and then restored afterwards?